apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.ExecutableCommand.cs

```diff
@@ -311,17 +311,10 @@ private async Task<int> ExecuteMutationsAsync(CancellationToken cancellationToke
 
             private List<Mutation> GetMutations()
             {
-                // Currently, ToProtobufValue doesn't use the options it's provided. They're only
-                // required to prevent us from accidentally adding call sites that wouldn't be able to obtain
-                // valid options. For efficiency, we just pass in null for now. If we ever need real options
-                // from the connection string, uncomment the following line to initialize the options from the connection.
-                // SpannerConversionOptions options = SpannerConversionOptions.ForConnection(SpannerConnection);
-                SpannerConversionOptions conversionOptions = null;
-
                 // Whatever we do with the parameters, we'll need them in a ListValue.
                 var listValue = new ListValue
                 {
-                    Values = { Parameters.Select(x => x.SpannerDbType.ToProtobufValue(x.GetValidatedValue(), conversionOptions)) }
+                    Values = { Parameters.Select(x => x.GetConfiguredSpannerDbType(ConversionOptions).ToProtobufValue(x.GetValidatedValue(), ConversionOptions)) }
```

Comment:
Is it worth evaluating ConversionOptions once before the implicit loop here? Not sure. (We can definitely optimize later if we really need to.)

Reply:
Good catch. I will make this change.

Reply:
Now that we have made it lazy, so options construction should happen only once. We should be good now?